### DISCUSSION TITLE
How to avoid accidental connections to incompatible federations?

### DISCUSSION THREAD

**Discussion by justinmoon** [fedimint/fedimint#1329] [General]
The other day I was trying to peg-out of a test federation that I set up.

I accidentally attempted the peg-out using a client that's not the same as the federation was running. The commit was from the same day, but wasn't exactly the same commit. I hit an error and lost funds.

The real fix here is a modularized client that can't lose funds (ongoing), but perhaps we want to prevent incompatible client & server from even attempting to work together.

**Comment by dpc** [comment]
Connection greeting command with federation id in it? Could be empty `Option::None`during setup.

**Comment by elsirion** [comment]
Once we have versioning there could be a version check when opening an API connection. Before that it's kinda futile imo. We need these versions asap.
